src/ol_openedx_course_sync/ol_openedx_course_sync/utils.py

```diff
@@ -136,3 +149,135 @@ def update_default_tabs(source_course_key, target_course_key, user):
 
     if is_updated:
         store.update_item(target_course, user.id)
+
+
+def sync_discussions_configuration(source_course_key, target_course_key, user):
+    """
+    Sync DiscussionsConfiguration and CourseDiscussionSettings
+    from source to target course.
+
+    Args:
+        source_course_key (CourseKey): The key for the source course.
+        target_course_key (CourseKey): The key for the target course.
+    """
+
+    def sync_model_objects(source_object, target_object, fields_to_sync):
+        has_changes = False
+        for field in fields_to_sync:
+            if getattr(source_object, field) != getattr(target_object, field):
+                has_changes = True
+                setattr(target_object, field, getattr(source_object, field))
+        if has_changes:
+            target_object.save()
+        return has_changes
+
+    fields_to_sync = {
+        # `DiscussionsConfiguration` fields excluding context_key and history.
+        DiscussionsConfiguration: [
+            "enabled",
+            "posting_restrictions",
+            "lti_configuration",
+            "enable_in_context",
+            "enable_graded_units",
+            "unit_level_visibility",
+            "plugin_configuration",
+            "provider_type",
+        ],
+        # `CourseDiscussionSettings` fields excluding course_id and discussions_id_map.
+        CourseDiscussionSettings: [
+            "always_divide_inline_discussions",
+            "reported_content_email_notifications",
+            "division_scheme",
+            "_divided_discussions",
+        ],
+    }
+
+    source_discussions_settings = CourseDiscussionSettings.get(source_course_key)
+    target_discussions_settings = CourseDiscussionSettings.get(target_course_key)
+    _ = sync_model_objects(
+        source_discussions_settings,
+        target_discussions_settings,
+        fields_to_sync[CourseDiscussionSettings],
+    )
+
+    source_discussions_config = DiscussionsConfiguration.get(source_course_key)
+    target_discussions_config = DiscussionsConfiguration.get(target_course_key)
+    has_configuration_changes = sync_model_objects(
+        source_discussions_config,
+        target_discussions_config,
+        fields_to_sync[DiscussionsConfiguration],
+    )
+
+    if not has_configuration_changes:
+        return
+
+    # update discussion settings in modulestore
+    source_course = modulestore().get_course(source_course_key)
+    target_course = modulestore().get_course(target_course_key)
+
+    target_course.discussions_settings["enable_in_context"] = (
```

Comment:
If target_course is None, the code below will break.

Reply:
The target course can never be None because our model CourseSyncMapping already checks for that before creating a mapping.